### PR TITLE
fix(storage): limit shared buffer compact sst size

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

We found that in the previous design, the size of all imm was used as the compact size.
And, rely on the ability of the `vnode_partition` to split sst.

This PR, add the limit to avoid generating the huge size `Sstable`

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by hzxa21** [comment]
Please add some background. Maybe we should create an issue for this.

File: src/storage/src/hummock/compactor/shared_buffer_compact.rs [hunk 1]
```diff
@@ -552,7 +552,7 @@ fn generate_splits(
                 split_weight_by_vnode /= 2;
                 avg_vnode_size *= 2;
             }
-            sub_compaction_sstable_size = compact_data_size;
+            sub_compaction_sstable_size = std::cmp::min(sstable_size, compact_data_size * 6 / 5);
```
Comment (hzxa21) [inline src/storage/src/hummock/compactor/shared_buffer_compact.rs:555]: how about moving L513 to L560 instead of adding this line.

**Review by MrCroxx — COMMENTED** [review]
LGTM, and please add some extra useful comments about the other part of the function. 🥰

**Comment by hzxa21** [comment]
> LGTM, and please add some extra useful comments about the other part of the function. 🥰

+1 🥰

**Review by hzxa21 — APPROVED** [review]

**Review by zwang28 — APPROVED** [review]

**Pull request merged**